kafka/integration_test.go

```diff
@@ -3235,6 +3235,85 @@ func (its *IntegrationTestSuite) TestAdminClient_ListOffsets() {
 
 }
 
+// Test DeleteRecords API which deletes all the records before the specified offset
+// in the particular partition of the specified topic.
+func (its *IntegrationTestSuite) TestAdminClient_DeleteRecords() {
+    t := its.T()
+	bootstrapServers := testconf.Brokers
+	assert := its.Assert()
+
+	// Create a new AdminClient.
+	a := createAdminClient(t)
+	defer a.Close()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+    
+	// Create a new topic to test the api and produce some messages to that topic
+	Topic := "test-delete"
+    
+	topics := []TopicSpecification{TopicSpecification{Topic: Topic, NumPartitions: 1, ReplicationFactor: 1}}
+	createTopicResult, createTopicError := a.CreateTopics(ctx, topics)
+	assert.Nil(createTopicError, "Create Topics should not fail.")
+	assert.Equal(createTopicResult[0].Error.Code(), ErrNoError, "Create Topics Error Code should be ErrNoError.")
+
+    p, err := NewProducer(&ConfigMap{"bootstrap.servers": bootstrapServers})
+	assert.Nil(err, "Unable to create Producer.")
+	defer p.Close()
+
+	p.Produce(&Message{
+		TopicPartition: TopicPartition{Topic: &Topic, Partition: 0},
+		Value:          []byte("Message-1"),
+	}, nil)
+
+	p.Produce(&Message{
+		TopicPartition: TopicPartition{Topic: &Topic, Partition: 0},
+		Value:          []byte("Message-2"),
+	}, nil)
+
+	p.Produce(&Message{
+		TopicPartition: TopicPartition{Topic: &Topic, Partition: 0},
+		Value:          []byte("Message-3"),
+	}, nil)
+
+	p.Flush(5 * 1000)
+    
+	// Delete the records till offset 2 in partion 0 of the topic
+	// The result will contain the minimum offset available after deletion in that Topic Partiton
+	var delRecordsTopicPartitionOffsets []TopicPartition
+	delRecordsTopicPartitionOffsets = append(delRecordsTopicPartitionOffsets, TopicPartition{
+		Topic:     &Topic,
+		Partition: int32(0),
+		Offset:    Offset(2),
+	})
+	var deleteRes DeleteRecordsResult
+	deleteRes, err = a.DeleteRecords(ctx,delRecordsTopicPartitionOffsets)
+	assert.Nil(err,"Delete Records should not fail")
+	var offsetAfterDeletion Offset
+	for _,tp := range deleteRes.TopicPartitions {
```

Comment:
Rather than this, assert here that the length of deleteRes.TopicPartitions is 1, assert that deleteRes.TopicPartitions[0] has error not set, and finally, just use deleteRes.TopicPartitions[0].Offset